Userland/Libraries/LibGUI/Window.cpp

```diff
@@ -522,6 +533,25 @@ void Window::propagate_shortcuts(KeyEvent& event, Widget* widget, ShortcutPropag
     event.ignore();
 }
 
+void Window::remember_state(DeprecatedString const& domain, DeprecatedString const& group)
```

Comment:
Reading this function name from the call site, I'd have no idea what "state" is. Let's call it something meaningful like `restore_size_and_position`.

It also seems like this should take a default rect, something like:
```c++
void Window::restore_size_and_position(StringView domain, StringView group, Gfx::IntRect default_rect)
{
    ...

    int x = Config::read_i32(domain, group, "X"sv, rect.x());
    int y = Config::read_i32(domain, group, "Y"sv, rect.y());
    int width = Config::read_i32(domain, group, "Width"sv, rect.width());
    int height = Config::read_i32(domain, group, "Height"sv, rect.height());
    set_rect({ x, y, width, height });

    ....
}
```

Reply:
You are right it was way to vague, I have updated the functions to be mare clear.